### PR TITLE
Add basic support for plotting preshaped qcodes data directly

### PR DESCRIPTION
**Pull request by jenshnielsen**
This will require https://github.com/QCoDeS/Qcodes/pull/2264 and qcodes 0.20.0 release to actually work

### CODE REVIEW

File: test/pytest/test_qcodes_flow.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        ('Subtract average', SubtractAverage),
+        ('plot', PlotNode)
+    )
+    loader = fc.nodes()['Data loader']
+    selector = fc.nodes()['Data selection']
+    selector.selectedData = 'z_0'
+
+    loader.pathAndId = dataset_with_shape.path_to_db, dataset_with_shape.run_id
+    loader.update()
+    assert 1 == 1
```
Comment (jenshnielsen) [inline test/pytest/test_qcodes_flow.py]: This test does need a bit more work

**Review by astafan8 — APPROVED** [review]
i think this is good :) it was tough to review though because even though the changes are minimal, it does take time to understand what happens where and when and where shape is actually used, etc. :)

**Comment by astafan8** [comment]
it seems to me that `datadict_to_meshgrid` could use some optimization/improvement so that it does not recreate arrays if existing shape is to be used. but i don't know if i'm even correct. and it's definitely not for this PR.

File: test/pytest/test_qcodes_flow.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+@pytest.mark.skipif(version.parse(qc.__version__)
+                    < version.parse("0.20.0"),
+                    reason="Requires QCoDes 0.20.0 or later")
+def test_qcodes_flow_shaped_data(qtbot, dataset_with_shape):
+
+    fc = linearFlowchart(
+        ('Data loader', QCodesDSLoader),
+        ('Data selection', DataSelector),
+        ('Grid', DataGridder),
+        # ('Dimension assignment', XYSelector),
```
Comment (jenshnielsen) [inline test/pytest/test_qcodes_flow.py]: Remove this commented out line

**Pull request merged**